docs/guide/controllers.md

```diff
@@ -259,6 +348,71 @@ Results from the search are assigned to the `blog_posts` variable as a `SearchRe
 {% endfor %}
 ```
 
+### Creating custom Query Types
+
+Beyond the built-in Query Types, you can create your own.
+
+For example, this Query Type returns a Query that searches for the 10 last published Content items, ordered by reverse publishing date.
+It accepts an optional `type` parameter that can be set to a Content Type identifier:
+
+``` php
+<?php
+namespace App\QueryType;
+
+use eZ\Publish\Core\QueryType\QueryType;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+
+class LatestContentQueryType implements QueryType
+{
+    public function getQuery(array $parameters = [])
+    {
+        $criteria[] = new Query\Criterion\Visibility(Query\Criterion\Visibility::VISIBLE);
+        if (isset($parameters['type'])) {
+            $criteria[] = new Query\Criterion\ContentTypeIdentifier($parameters['type']);
+        }
+        // 10 is the default limit we set, but you can have one defined in the parameters
+        return new LocationQuery([
+            'filter' => new Query\Criterion\LogicalAnd($criteria),
+            'sortClauses' => [new Query\SortClause\DatePublished()],
+            'limit' => isset($parameters['limit']) ? $parameters['limit'] : 10,
+        ]);
+    }
+    public static function getName()
+    {
+        return 'LatestContent';
+    }
+    /**
+     * Returns an array listing the parameters supported by the QueryType.
+     * @return array
+     */
+    public function getSupportedParameters()
+    {
+        return ['type', 'limit'];
+    }
+}
+```
+
+In the configuration, use the name of the Query Types as defined in `getName`.
```

Comment:
In what file?

Reply:
Can be different files (ezplatform.yaml by default, but not necessarily), so I clarified that it's content view config.